test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/OpenShiftApplicationExplorerView.java

```diff
@@ -0,0 +1,155 @@
+		new DefaultShell(OpenShiftLabel.Shell.LOGIN);
+	}
+
+	public void connectToOpenShiftODO() {
+		switch (DatastoreOS3.AUTH_METHOD) {
+		case BASIC:
+			connectToOpenShiftODOBasic(DatastoreOS3.SERVER, DatastoreOS3.USERNAME, DatastoreOS3.PASSWORD);
+			break;
+		case OAUTH:
+			connectToOpenShiftODOOAuth(DatastoreOS3.SERVER, DatastoreOS3.TOKEN);
+			break;
+		default:
+			throw new OpenShiftToolsException("no valid authentication method found. Could not connect.");
+		}
+	}
+
+	/**
+	 * Connects to OpenShift server. OpenShift connection shell has to be opened at
+	 * the moment of method invocation.
+	 * 
+	 * @param server           URL of a server
+	 * @param username
+	 * @param password
+	 */
+	public void connectToOpenShiftODOBasic(String server, String username, String password) {
+		connectToOpenShift(server, username, password, null);
+	}
+
+	/**
+	 * Connects to OpenShift server. OpenShift connection shell has to be opened at
+	 * the moment of method invocation.
+	 * 
+	 * @param server           URL of a server
+	 * @param token
+	 */
+	public void connectToOpenShiftODOOAuth(String server, String token) {
+		connectToOpenShift(server, null, null, token);
+	}
+
+	public void connectToOpenShift(String server, String username, String password, String token) {
+		openConnectionShell();
+		new LabeledText(OpenShiftLabel.TextLabels.URL).setText(server);
+
+		if (StringUtils.isEmpty(username) && StringUtils.isEmpty(password) && !StringUtils.isEmpty(token)) {
+			new LabeledText(OpenShiftLabel.TextLabels.TOKEN_ODO).setText(token);
+		} else if (!StringUtils.isEmpty(username) && !StringUtils.isEmpty(password) && StringUtils.isEmpty(token)) {
+			new LabeledText(OpenShiftLabel.TextLabels.USERNAME).setText(username);
+			new LabeledText(OpenShiftLabel.TextLabels.PASSWORD).setText(password);
+		} else {
+			throw new OpenShiftToolsException("Wrong or incomplete credentials!");
+		}
+
+		new WaitUntil(new ControlIsEnabled(new FinishButton()), TimePeriod.DEFAULT);
+
+		new FinishButton().click();
+
+		new WaitWhile(new ShellIsAvailable(OpenShiftLabel.Shell.LOGIN), TimePeriod.LONG);
+		new WaitWhile(new JobIsRunning(), TimePeriod.LONG);
+	}
+
+	public boolean connectionExistsAndWorking() {
+		try {
+			OpenShiftODOConnection connection = getOpenShiftODOConnection();
+			connection.expand();
+			return true;
+		} catch (Exception ex) {
+			return false;
+		}
+	}
+
+	/**
+	 * Finds out whether connection with specified username and server exists or
+	 * not.
+	 * 
+	 * @param server   server
+	 * @return true if connection exists, false otherwise
+	 */
+	public boolean connectionExists() {
+		try {
+			return getConnectionItem() != null;
+		} catch (RedDeerException ex) {
+			return false;
+		}
+	}
+
+	public OpenShiftODOConnection getOpenShiftODOConnection() {
+		OpenShiftODOConnection connection = null;
+
+		connection = new OpenShiftODOConnection(getConnectionItem());
+
+		assertNotNull("Unable to get OpenShift connection.", connection);
```

Comment:
Why to use assert in test framework? Wouldn't it cleaner to use throwing an exception and deal with it on the usage, ie. line: 121?